libraries/Microsoft.Bot.Connector/ConnectorClientEx.cs

```diff
@@ -62,6 +62,42 @@ public ConnectorClient(Uri baseUri, MicrosoftAppCredentials credentials, HttpCli
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ConnectorClient"/> class.
+        /// </summary>
+        /// <param name="baseUri">Base URI for the Bot Connector service.</param>
+        /// <param name="credentials">Credentials for the Bot Connector service.</param>
+        /// <param name="customHttpClient">The HTTP client to use for this connector client.</param>
+        /// <param name="disposeHttpClient">Whether to dispose the <see cref="HttpClient"/>.</param>
+        /// <remarks>Constructor specifically designed to be the one that allows control of the disposing of the custom <see cref="HttpClient"/>.
+        /// <see cref="ServiceClient{T}"/> only has one constructor that accepts control of the disposing of the <see cref="HttpClient"/>, so we call that overload here.
+        /// All other overloads of <see cref="ConnectorClient"/> will not control this parameter and it will default to true, resulting on disposal of the provided <see cref="HttpClient"/> when the <see cref="ConnectorClient"/> is disposed.
+        /// When reusing <see cref="HttpClient"/> instances across connectors, pass 'false' for <paramref name="disposeHttpClient"/> to avoid <see cref="ObjectDisposedException"/>.</remarks>
+#pragma warning disable CA1801 // Review unused parameters (we can't change this without breaking binary compat)
+        public ConnectorClient(Uri baseUri, ServiceClientCredentials credentials, HttpClient customHttpClient, bool disposeHttpClient)
+#pragma warning restore CA1801 // Review unused parameters
+            : base(customHttpClient, disposeHttpClient)
+        {
+            this.Credentials = credentials;
+            if (customHttpClient != null)
+            {
+                this.HttpClient = customHttpClient;
```

Comment:
Is this needed now? since we are passing the custom httpClient to the base class?

IF we are doing this, then shouldn't we dispose the underlying HttpClient before we stomp on it? or it will be orphaned

Reply:
Yeah this is not needed. However it was just assigning the same twice so there were no memory problems. It was however redundant.